### DISCUSSION TITLE
Name of function is cleared after val_graph optimization

### DISCUSSION THREAD

**Discussion by metab0t** [coin-or/CppAD#225] [Q&A]
The name of function is reset after `val2fun` but is unchanged if we do not use `val_graph` optimization. Should the behavior be consistent across different optimization options?
https://github.com/coin-or/CppAD/blob/7a81cb7a777ad1536ea3bfdaf5249eae33ffd5ed/include/cppad/local/val_graph/val2fun.hpp#L1000

**Comment by bradbell** [comment]
The val2fun optimization is experimental. It is much easier to understand, but it take a lot of time to convert from and to a val_graph representation. 

I plan to make an optimizer that is simpler, like val_graph, but not slower.
